src/constraints/date-in-past.js

```diff
@@ -0,0 +1,19 @@
+import { Literal } from "rdflib";
+import { XSD } from "../namespaces.js";
+
+/**
+ * @param {Literal} dateLiteral
+ * @returns {boolean}
+ */
+export default function dateInPast(dateLiteral /*, options*/) {
+  if (
+    !dateLiteral.datatype.equals(XSD("date")) &&
+    !dateLiteral.datatype.equals(XSD("dateTime"))
+  ) {
+    return false;
+  }
+
+  const date = Literal.toJS(dateLiteral);
+
+  return date <= Date.now();
```

Comment:
Pretty basic implementation, I wasn't sure if I needed to check the validity of the dates, since we have the other constraints for that as well.